### PR TITLE
[AzureMonitorExporter] refactor logging (part4) - test helper

### PR DESCRIPTION
**Pull request by TimothyMothra**
Towards #34581

Need to have adequate test coverage of all the new methods being added.

### Changes
- Add new classes `EventSourceTestHelper` and `TestEventListener`

### CODE REVIEW

**Comment by TimothyMothra** [comment]
Converting to draft while I investigate the correct way to cite where this was copied from.

**Comment by rajkumar-rangaraj** [comment]
Could you add a detailed description on the importance of this test. From the description I understand it covers test gaps but information related to why event source needs to be tested could help.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/CommonTestFramework/TestEventListener.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            this.OnOnEventWritten(eventData);
+        }
+    }
+
+    /// <summary>Handler for event source creation.</summary>
+    /// <param name="eventSource">The event source that was created.</param>
+    protected override void OnEventSourceCreated(EventSource eventSource)
+    {
+        // Check for null because this method is called by the base class constructor before we can initialize it
+        Action<EventSource>? callback = this.OnOnEventSourceCreated;
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/CommonTestFramework/TestEventListener.cs:84]: where is `OnOnEventSourceCreated` initialized?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/CommonTestFramework/EventSourceTestHelper.cs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Tests.CommonTestFramework;
+
+internal static class EventSourceTestHelper
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/CommonTestFramework/EventSourceTestHelper.cs:13]: add a comment linking to Otel, if this inspired by OTel tests.!

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/CommonTestFramework/EventSourceTestHelper.cs:13]: I reached out to CELA last week and they didn't think a citation was necessary. I'll double check just to confirm we're doing the right thing here.

It's worth noting though that OTel copied these classes from Application Insights (without citation).
Who would you recommend we cite here - OTel or AI?

Comment (cijothomas) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/CommonTestFramework/EventSourceTestHelper.cs:13]: i wasn't commenting based on the legal aspects! Purely for the purpose of fixing the original one if something new was discovered here!

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/CommonTestFramework/TestEventListener.cs (hunk 1 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/CommonTestFramework/TestEventListener.cs:84]: wow good catch! 
it's in the original AI version of the class, but I don't see where it's ever used.

**Review by vishweshbankwar — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/CommonTestFramework/EventSourceTestHelper.cs (hunk 2 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/CommonTestFramework/EventSourceTestHelper.cs:16]: added citations.

**Pull request merged**